src/lsp/dune

```diff
@@ -0,0 +1,5 @@
+(library
```

Comment:
As all LSP files are prefixed by lsp_, why not putting every thing in src/ and remove src/lsp/?

Reply:
That's what I wanted to do at first, but for some reason there already is a `pure` directory. So why not split the files among several directories? We are starting to get many files in there anyway.

Reply:
It is not very convenient... pure.ml could be moved in src/ too... I feel that it is still manageable but I would be happy to hear other opinions.

Reply:
Let's see what @ejgallego has to say on that. Personally I don't mind flattening the directories and removing all the library stuff we build using dune to only produce the main executable. But in that case, what should we do with Emilio's change log and README?

Reply:
We could postfix them by LSP.

Reply:
I would rather move the contents to the doc, and extend the main README with minimal information on how to use the LSP server. For the change log, I would simply remove it, or turn it into a change log for the whole project.